### PR TITLE
Support Azure blob as source

### PR DESCRIPTION
**Pull request by summerhasama**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

**Review by NarayanThiru — CHANGES_REQUESTED** [review]

File: src/k8s-configuration/azext_k8s_configuration/consts.py [hunk 1]
```diff
@@ -41,7 +41,30 @@
 REQUIRED_BUCKET_VALUES_MISSING_HELP = (
     "Provide either both of '--secret-key' and '--access-key' or '--local-auth-ref'"
 )
-
+REQUIRED_AZURE_BLOB_SERVICE_PRINCIPAL_VALUES_MISSING_ERROR = (
+    "Error! Service principal is invalid because it is missing value(s)"
+)
+REQUIRED_AZURE_BLOB_SERVICE_PRINCIPAL_VALUES_MISSING_HELP = (
+    "Provide '--sp-client-id', '--sp-tenant-id', and one of '--sp-client-secret' or '--sp-client-cert'"
```
Comment (bavneetsingh16) [inline src/k8s-configuration/azext_k8s_configuration/consts.py:48]: replace 'one of' with 'either'

File: src/k8s-configuration/azext_k8s_configuration/consts.py [hunk 2]
```diff
@@ -41,7 +41,30 @@
...
+)
+REQUIRED_AZURE_BLOB_SERVICE_PRINCIPAL_AUTH_ERROR = (
+    "Error! Too many authentication methods for service principal"
+)
+REQUIRED_AZURE_BLOB_SERVICE_PRINCIPAL_AUTH_HELP = (
+    "Provide only one of '--sp-client-secret' or '--sp-client-cert'"
```
Comment (bavneetsingh16) [inline src/k8s-configuration/azext_k8s_configuration/consts.py:54]: replace with 'either'

File: src/k8s-configuration/azext_k8s_configuration/consts.py [hunk 3]
```diff
@@ -41,7 +41,30 @@
...
+)
+REQUIRED_AZURE_BLOB_SERVICE_PRINCIPAL_CERT_VALUES_MISSING_ERROR = (
+    "Error! Service principal certificate password is invalid"
+)
+REQUIRED_AZURE_BLOB_SERVICE_PRINCIPAL_CERT_VALUES_MISSING_HELP = (
+    "Provide '--sp-client-id', '--sp-tenant-id', and '--sp-client-cert' with your '--sp-cert-password"
+)
+REQUIRED_AZURE_BLOB_AUTH_ERROR = (
+    "Error! Too many authentication methods for Azure Blob"
```
Comment (bavneetsingh16) [inline src/k8s-configuration/azext_k8s_configuration/consts.py:63]: 'provided'

File: src/k8s-configuration/azext_k8s_configuration/consts.py [hunk 4]
```diff
@@ -41,7 +41,30 @@
...
+)
+REQUIRED_AZURE_BLOB_AUTH_HELP = (
+    "Specify one authentication method of '--local-auth-ref', '--account-key', '--sas-token', '--mi-client-id', or service principal with '--sp-client-id', '--sp-tenant-id', and one of '--sp-client-secret' or '--sp-client-cert'"
```
Comment (bavneetsingh16) [inline src/k8s-configuration/azext_k8s_configuration/consts.py:66]: Specify one of the available authentication methods from the list:

File: src/k8s-configuration/azext_k8s_configuration/_params.py [hunk 5]
```diff
@@ -178,6 +178,62 @@ def load_arguments(self, _):
             help="Define kustomizations to sync sources with parameters ['name', 'path', 'depends_on', 'timeout', 'sync_interval', 'retry_interval', 'prune', 'force']",
             nargs="+",
         )
+        c.argument(
+            "container_name",
+            help="Name of the Azure Blob Storage container to sync",
```
[the remark this answers is not in the document]
Comment (bavneetsingh16) [reply src/k8s-configuration/azext_k8s_configuration/_params.py:183]: got it reviewed

File: src/k8s-configuration/azext_k8s_configuration/consts.py [hunk 6]
```diff
@@ -41,7 +41,30 @@
 REQUIRED_BUCKET_VALUES_MISSING_HELP = (
     "Provide either both of '--secret-key' and '--access-key' or '--local-auth-ref'"
 )
-
+REQUIRED_AZURE_BLOB_SERVICE_PRINCIPAL_VALUES_MISSING_ERROR = (
```
[the remark this answers is not in the document]
Comment (bavneetsingh16) [reply src/k8s-configuration/azext_k8s_configuration/consts.py:44]: got it reviewed and made changes as suggested

**Review by NarayanThiru — APPROVED** [review]

**Pull request merged**